admiral/pkg/clusters/serviceentry.go

```diff
@@ -802,28 +801,26 @@ func modifyServiceEntryForNewServiceOrPod(
 		ctxLogger.Infof(common.CtxLogFormat, "updateRegistryConfigForClusterPerEnvironment", deploymentOrRolloutName, deploymentOrRolloutNS, "", "done writing")
 		return nil, nil
 	}
-	// If VS based routing is enabled, then generate VirtualServices for the source cluster's ingress
-	// This is done after the ServiceEntries are created for the source cluster
-	if common.IsVSBasedRoutingEnabled() {
-		// Writing phase: We update the base ingress virtualservices with the RouteDestinations
-		// gathered during the discovery phase and write them to the source cluster
-		err := addUpdateVirtualServicesForSourceIngress(ctx, ctxLogger, remoteRegistry, sourceClusterToDestinations)
```

Comment:
Should we rename to `WriteVSForIngress`?
Is `SourceIngress` or `DestinationIngress` of any concern?

Also why do we have global functions? Shouldn't a source cluster have this API to write to a destination?

Reply:
changing the name to `addUpdateVirtualServicesForIngress`
This is not exactly a global func, this func is accessible from within the package.

Reply:
true. to clarify i meant it should be been a method on a specific struct on the source cluster. but it seems we dont have the struct at this time.